quarkc/parser.py

```diff
@@ -318,8 +318,8 @@ def visit_types(self, node, (first, rest)):
     def visit_name_re(self, node, children):
         return node.text
 
-    @g.rule('name = _ name_re  _')
-    def visit_name(self, node, (pre, name, post)):
+    @g.rule('name = _ !(keyword !~"[_a-zA-Z0-9]") name_re  _')
+    def visit_name(self, node, (pre, _, name, post)):
         return Name(name)
 
```

Comment:
@bozzzzo we could try doing that and see if it works. In general, if we do not alter the grammar, then we accept a different language in the first place, and I don't know if checking identifiers in a separate pass would lead to the same effect, or if the effect (on what language is accepted) would be different.